tests/expr_and_series/rolling_sum_test.py

```diff
@@ -0,0 +1,197 @@
+from __future__ import annotations
+
+from typing import Any
+
+import pytest
+
+import narwhals.stable.v1 as nw
+from narwhals.exceptions import InvalidOperationError
+from tests.utils import Constructor
+from tests.utils import ConstructorEager
+from tests.utils import assert_equal_data
+
+data = {"a": [None, 1, 2, None, 4, 6, 11]}
+
+kwargs_and_expected = {
+    "x1": {"kwargs": {"window_size": 3}, "expected": [float("nan")] * 6 + [21]},
+    "x2": {
+        "kwargs": {"window_size": 3, "min_periods": 1},
+        "expected": [float("nan"), 1.0, 3.0, 3.0, 6.0, 10.0, 21.0],
+    },
+    "x3": {
+        "kwargs": {"window_size": 2, "min_periods": 1},
+        "expected": [float("nan"), 1.0, 3.0, 2.0, 4.0, 10.0, 17.0],
+    },
+    "x4": {
+        "kwargs": {"window_size": 5, "min_periods": 1, "center": True},
+        "expected": [3.0, 3.0, 7.0, 13.0, 23.0, 21.0, 21.0],
+    },
+    "x5": {
+        "kwargs": {"window_size": 4, "min_periods": 1, "center": True},
+        "expected": [1.0, 3.0, 3.0, 7.0, 12.0, 21.0, 21.0],
+    },
+    "x6": {
+        "kwargs": {"window_size": 0},
+        "expected": [float("nan"), 1.0, 3.0, 3.0, 7.0, 13.0, 24.0],
+    },
+    # There are still some edge cases to take care of with nulls and min_periods=0:
```

Comment:
Example:

```py
In [1]: import pandas as pd

In [2]: import polars as pl

In [3]: data = [float("nan"), 1, 2]

In [4]: pl.Series(data).rolling_sum(2, min_periods=0)
Out[4]: 
shape: (3,)
Series: '' [f64]
[
        NaN
        NaN
        3.0
]

In [5]: pd.Series(data).rolling(2, min_periods=0).sum()
Out[5]: 
0    0.0
1    1.0
2    3.0
dtype: float64
```

Reply:
Honestly, this seems buggy in both?

- polars: why the second value is NaN if `min_periods=0`?
- pandas: why replace the first value with a 0?

Reply:
in Polars `min_periods` refers to missing data, not to `'nan'`

the mean of `'nan'` and 1 is '`nan'`